### PR TITLE
kiwix#2761 converted KiwixIdlingResource.java to kotlin

### LINKED ISSUE #2761: Convert KiwixIdlingResource.java to kotlin
parent issue #1826

### What changes you will do:
I will convert KiwixIdlingResource.java file to Kotlin

### PR DESCRIPTION
**Pull request by cdhiraj40**
fixes #2761

### CODE REVIEW

File: app/src/androidTest/java/org/kiwix/kiwixmobile/utils/KiwixIdlingResource.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  companion object {
+    private var kiwixIdlingResource: KiwixIdlingResource? = null
+
+    @JvmStatic
+    fun getInstance(): KiwixIdlingResource? {
+      if (kiwixIdlingResource == null) {
+        kiwixIdlingResource = KiwixIdlingResource()
+      }
+      kiwixIdlingResource!!.idle = true
```
Comment (gouri-panda) [inline app/src/androidTest/java/org/kiwix/kiwixmobile/utils/KiwixIdlingResource.kt:41]: Probably we should check this in if the condition. We don't know how long it takes to instantiate.

File: app/src/androidTest/java/org/kiwix/kiwixmobile/utils/KiwixIdlingResource.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  override fun startTask() {
+    idle = false
+  }
+
+  override fun finishTask() {
+    idle = true
+    if (resourceCallback != null) {
+      resourceCallback!!.onTransitionToIdle()
+    }
```
Comment (gouri-panda) [inline app/src/androidTest/java/org/kiwix/kiwixmobile/utils/KiwixIdlingResource.kt:63]: [nit] resourceCallback?.onTransitionToIdle()

File: app/src/androidTest/java/org/kiwix/kiwixmobile/utils/KiwixIdlingResource.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+/**
+ * Created by mhutti1 on 19/04/17.
+ */
+
+class KiwixIdlingResource : IdlingResource, IdleListener {
+  private var idle = true
+  private var resourceCallback: ResourceCallback? = null
+
+  companion object {
```
Comment (gouri-panda) [inline app/src/androidTest/java/org/kiwix/kiwixmobile/utils/KiwixIdlingResource.kt:33]: We generally put companion objects at the end of the files.

File: app/src/androidTest/java/org/kiwix/kiwixmobile/utils/KiwixIdlingResource.kt [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  companion object {
+    private var kiwixIdlingResource: KiwixIdlingResource? = null
+
+    @JvmStatic
+    fun getInstance(): KiwixIdlingResource? {
+      if (kiwixIdlingResource == null && kiwixIdlingResource!!.idle) {
+        kiwixIdlingResource = KiwixIdlingResource()
+      }
```
Comment (gouri-panda) [inline app/src/androidTest/java/org/kiwix/kiwixmobile/utils/KiwixIdlingResource.kt:59]: This'll  crash. Suppose if `kiwixIdlingResource`is null. Howcome it has `idle` property? Also you're forgetting this
```
kiwixIdlingResource.idle = true
```

Comment (cdhiraj40) [reply app/src/androidTest/java/org/kiwix/kiwixmobile/utils/KiwixIdlingResource.kt:59]: yes, that does make sense, my bad. I will update PR soon :)

File: app/src/androidTest/java/org/kiwix/kiwixmobile/utils/KiwixIdlingResource.kt [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  companion object {
+    private var kiwixIdlingResource: KiwixIdlingResource? = null
+
+    @JvmStatic
+    fun getInstance(): KiwixIdlingResource? {
+      if (kiwixIdlingResource == null) {
+        kiwixIdlingResource = KiwixIdlingResource()
+        kiwixIdlingResource!!.idle = true
```
Comment (gouri-panda) [inline app/src/androidTest/java/org/kiwix/kiwixmobile/utils/KiwixIdlingResource.kt:59]: We need both cases i.e null or nonnull.
```
if (kiwixIdlingResource == null) {
        kiwixIdlingResource = KiwixIdlingResource()
}
if (kiwixIdlingResource == null) {
        kiwixIdlingResource!!.idle = true
}

**Comment by kelson42** [comment]
@gouri-panda Coukd you please domthe review?

**Review by gouri-panda — DISMISSED** [review]

**Pull request merged**